pandas/core/construction.py

```diff
@@ -513,10 +512,17 @@ def sanitize_masked_array(data: ma.MaskedArray) -> np.ndarray:
     """
     mask = ma.getmaskarray(data)
     if mask.any():
-        dtype, fill_value = maybe_promote(data.dtype, np.nan)
-        dtype = cast(np.dtype, dtype)
+        dtype = cast(np.dtype, data.dtype)
         data = ma.asarray(data.astype(dtype, copy=True))
         data.soften_mask()  # set hardmask False if it was True
+
+        if np.issubdtype(dtype, np.integer):
+            fill_value: int | float | None = np.iinfo(dtype).min
```

Comment:
When a value in the input is masked, it is not correct to fill in with the minimum value.

Reply:
@rhshadrach  I’ve been thinking about how to choose the right fill_value for integers in this situation. Since there’s no equivalent to NaN for integers, I’m struggling to find a good default value. I also considered raising an error and asking the user to provide a fill_value, but that seems like it would require a lot of changes. I’m not sure what other approaches I could consider to solve this problem.

Reply:
@rhshadrach I’ve re-analyzed the issue and made some new changes. I’m not entirely sure if my changes are correct, so I would appreciate it if you could take a look. If there are still issues, please feel free to point them out. I’m eager to learn and continue making corrections. 

Thanks again for your attention and understanding.